src/components/AddQuestion.js

```diff
@@ -0,0 +1,64 @@
+import { Button, HStack, Input, useToast } from '@chakra-ui/react';
+import { useEffect, useState } from 'react';
+import { useAuthContext } from './AuthContextProvider';
+
+const AddQuestion = ({ addQuestion }) => {
+  const { questions, setQuestions } = useAuthContext();
+  const [questionIsAdded, setQuestionIsAdded] = useState(false);
+  const [content, setContent] = useState('');
+  const toast = useToast();
+  const getUniqueOrder = () => {
+    return Date.now() + (Math.random() * 100000).toFixed();
+  };
+
+  useEffect(async () => {
+    if (questionIsAdded) {
+      let newOrder = getUniqueOrder();
+      const question = {
+        text: content,
+        order: newOrder,
+      };
+      try {
+        const createdQuestion = await addQuestion(question);
+        setQuestions([...questions, createdQuestion.data]);
+        setContent('');
+        setQuestionIsAdded(false);
+      } catch (error) {
+        console.log(error);
+      }
+    }
+  }, [questionIsAdded]);
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+    if (!content) {
+      toast({
+        title: 'No content',
+
+        status: 'error',
+        duration: 2000,
+        isClosable: true,
+      });
+      return;
+    }
+    setQuestionIsAdded(true);
```

Comment:
You should put the logic from useEffect here, instead of changing a state variable that triggers a useEffect :)